unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_add_layer_success
+ * @brief               Test case of controller_screen_add_layer() where wl_resource_get_user_data() success, return an object
+ *                      and get_layer_from_id() success, return an object
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Mocking the get_layer_from_id() does return an object
+ *                      -# Calling the controller_screen_add_layer()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() not be called
+ */
+TEST_F(ControllerTests, controller_screen_add_layer_success)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviScreen, 1);
+    struct ivi_layout_layer *l_layoutLayer[1] = {(struct ivi_layout_layer *)0xFFFFFFFF};
+    SET_RETURN_SEQ(get_layer_from_id, l_layoutLayer, 1);
+
+    uint32_t l_layer_id = 10;
+    controller_screen_add_layer(nullptr, nullptr, l_layer_id);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 0);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_remove_layer_nullScreen
+ * @brief               Test case of controller_screen_remove_layer() where wl_resource_get_user_data() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_remove_layer()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_remove_layer_nullScreen)
+{
+    uint32_t l_layer_id = 10;
+    controller_screen_remove_layer(nullptr, nullptr, l_layer_id);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_remove_layer_wrongLayerId
+ * @brief               Test case of controller_screen_remove_layer() where wl_resource_get_user_data() success, return an object
+ *                      but get_layer_from_id() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_screen_remove_layer()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_remove_layer_wrongLayerId)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviScreen, 1);
+
+    uint32_t l_layer_id = 10;
+    controller_screen_remove_layer(nullptr, nullptr, l_layer_id);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
```

Comment:
should verify wl_resource_post_event_fake.arg1_history and screen_remove_layer_fake.call_count